__tests__/unit/backend-proxy.test.js

```diff
@@ -282,5 +282,97 @@ describe('Backend Proxy', () => {
 				message: expect.stringContaining('Unexpected token')
 			}));
 		});
+
+		test('reads the backend response and makes the location a relative URL if the domain is the backend', () => {
```

Comment:
```suggestion
		test('forwards a 30x backend response with a rewritten location if the location header is to the backend server', () => {
```

the backend response doesn't get read when we `.pipe` it to the client